tools/distpartitioning/data_shuffle.py

```diff
@@ -273,29 +263,25 @@ def exchange_node_features(rank, world_size, node_feature_tids, ntype_gnid_map,
                 local_idx_partid = local_idx[cond]
 
                 if (gnids_per_partid.shape[0] == 0):
-                    node_feats_per_rank.append({feat_key : torch.empty((0,), dtype=torch.float)})
-                    global_nid_per_rank.append({feat_key : torch.empty((0,), dtype=torch.int64)})
+                    node_feats_per_rank.append(torch.empty((0,1), dtype=torch.float))
+                    global_nid_per_rank.append(np.empty((0,1), dtype=np.int64))
                 else:
-                    node_feats_per_rank.append({feat_key : node_feats[local_idx_partid]})
-                    global_nid_per_rank.append({feat_key : gnids_per_partid})
+                    if (len(list(node_feats[local_idx_partid].size())) == 1):
+                        node_feats_per_rank.append(node_feats[local_idx_partid])
+                    else:
+                        node_feats_per_rank.append(node_feats[local_idx_partid])
```

Comment:
what is the difference in the if branch and else branch?

Reply:
Agreed. This was the result of the assumption that treating vector of integers as a two dimensional vector and passing it to alltoallv, which was used prior to this fix. Since it is confirmed that this workaround does not work and the behavior is quite unpredictable this code became irrelevant now. But as a side effect the reshaping of vector to two dimensional array has been removed without checking the difference between the "if" case and "else" case in the above piece of code. 

Now we use alltoall for communication, this is no longer needed. And now we just treat the input data to alltoall in its original format and shape. Removing the "if" statement as suggested. 

Nice catch. Thanks so much.